flake8_trio.py

```diff
@@ -185,112 +188,153 @@ def visit_Import(self, node: ast.Import):
                 self.problems.append(make_error(TRIO106, node.lineno, node.col_offset))
 
 
+def critical_except(node: ast.ExceptHandler) -> Optional[Tuple[int, int, str]]:
+    def has_exception(node: Optional[ast.expr]) -> str:
+        if isinstance(node, ast.Name) and node.id == "BaseException":
+            return "BaseException"
+        if (
+            isinstance(node, ast.Attribute)
+            and isinstance(node.value, ast.Name)
+            and node.value.id == "trio"
+            and node.attr == "Cancelled"
+        ):
+            return "trio.Cancelled"
+        return ""
+
+    # bare except
+    if node.type is None:
+        return node.lineno, node.col_offset, "bare except"
+    # several exceptions
+    elif isinstance(node.type, ast.Tuple):
+        for element in node.type.elts:
+            name = has_exception(element)
+            if name:
+                return element.lineno, element.col_offset, name
+    # single exception, either a Name or an Attribute
+    else:
+        name = has_exception(node.type)
+        if name:
+            return node.type.lineno, node.type.col_offset, name
+    return None
+
+
 class Visitor102(Flake8TrioVisitor):
-    def __init__(self) -> None:
+    def __init__(self):
         super().__init__()
-        self._inside_finally: bool = False
-        self._scopes: List[TrioScope] = []
-        self._context_manager = False
+        self._critical_scope: Optional[Tuple[int, int, str]] = None
+        self._trio_context_managers: List[TrioScope] = []
+        self._safe_decorator = False
```

Comment:
I thought through these names again, and realized that both decorators and `with` are context managers, so I think these names are better.